src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs

```diff
@@ -139,5 +161,30 @@ private IEnumerable<IEdmStructuralProperty> ComputeDeclaredKey()
 
             return null;
         }
+
+        private IEnumerable<KeyValuePair<IEdmStructuralProperty, string>> ComputeDeclaredKeyAlias()
+        {
+            if (this.entity.Key != null)
+            {
+                List<KeyValuePair<IEdmStructuralProperty, string>> keyAlias = new List<KeyValuePair<IEdmStructuralProperty, string>>();
+                foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
+                {
+                    string propertyName = "";
+                    if (!String.IsNullOrEmpty(keyProperty.Alias))
+                    {
+                        string[] keyProperties = keyProperty.PropertyName.Split('/');
+                        propertyName = keyProperties[1];

```

Comment:
What if `propertyName` has multiple slashes?

Also, what if it does not have a slash at all, e.g. using an alias to an immediate property of the entity? This would throw an `IndexOutOfBoundException`. Or is there somewhere we're validating the the property name must include `/` if the `keyProperty.Alias` is non-empty?